dist/web3-light.js

```diff
@@ -13192,8 +13352,11 @@ module.exports = BigNumber; // jshint ignore:line
 
 },{}],"web3":[function(require,module,exports){
 var web3 = require('./lib/web3');
+
 web3.providers.HttpProvider = require('./lib/web3/httpprovider');
 web3.providers.QtSyncProvider = require('./lib/web3/qtsync');
+web3.providers.IpcProvider = require('./lib/web3/ipcprovider');
```

Comment:
eletctron has can preload files in a web view (type of iframe), which will have node integration, but are in a browser environment (e.g. having the window object).
In this stage i set the provider wrapper as IPC provider. The web page later loaded doesn't have the node integration and couldn't use the IPC, but even they could provide some kind of wrapper for an IPC/websocket behaviour, se even here the IPC provider makes sense, as its current behaviour very much matches the ones of web sockets too.

But for mist we need it available in the browser for sure and i don't see a problem having it visible to users.